.github/workflows/upload-cloud-images.yaml

```diff
@@ -153,4 +153,89 @@ jobs:
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - run: |
+          git fetch --prune --unshallow
+      # https://github.com/Azure/login?tab=readme-ov-file#azure-login-action
+      - name: Azure login
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Find latest stable version
+        run: |
+          # Azure only allows "stable" version strings. E.g. "v1.2.3" (not "v1.2.3-beta1")
+          latestTag=$(git tag --list | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' | sort -V | tail -n1)
+          echo $latestTag > LATEST_TAG
+
+      - name: Check if already pushed
+        id: checkPushed
+        run: |
+          latestTag=$(cat LATEST_TAG)
+
+          echo "Fetching all pushed versions"
+          mapfile -t kairosVersions < <(az sig image-version list --resource-group kairos-cloud-images --gallery-image-name kairos --gallery-name kairos.io --query '[].name' --output tsv)
+
+          echo "Checking if '$latestTag' is already pushed"
+          echo "Looking among versions: ${kairosVersions[@]}"
+          for version in "${kairosVersions[@]}"; do
+            if [[ $version == "${latestTag#v}" ]]; then
+              stableVersions+=("$version")
+              alreadyPushed=true
+              break
+            fi
+          done
+
+          if [[ "$alreadyPushed" = true && "${{ inputs.force }}" != "true" ]]; then
+            echo "shouldBuild=false" >> $GITHUB_OUTPUT
+            echo "Image for $latestTag is already pushed and 'force' wasn't true. Skipping build."
+          else
+            echo "shouldBuild=true" >> $GITHUB_OUTPUT
+            echo "Image for $latestTag is not pushed or 'force' was true. Will build."
+          fi
+
+      - name: Build the image
+        if: ${{ steps.checkPushed.outputs.shouldBuild == 'true' }}
+        run: |
+            latestTag=$(cat LATEST_TAG)
+            containerImage="quay.io/kairos/ubuntu:24.04-core-amd64-generic-${latestTag}"
+            docker run -v /var/run/docker.sock:/var/run/docker.sock --net host \
+              --privileged \
+              -v $PWD:/aurora --rm quay.io/kairos/auroraboot \
+              --debug \
+              --set "disable_http_server=true" \
+              --set "container_image=docker:${containerImage}" \
+              --set "disable_netboot=true" \
+              --set "disk.vhd=true" \
+              --set "state_dir=/aurora"
+
+      - name: Azure CLI script
+        uses: azure/cli@v2
+        if: ${{ steps.checkPushed.outputs.shouldBuild == 'true' }}
+        env:
+          GCP_PROJECT: palette-kairos
+          AZURE_RESOURCE_GROUP: "kairos-cloud-images"
+          AZURE_STORAGE_ACCOUNT: "kairoscloudimages"
+          AZURE_CONTAINER_NAME: "kairos-cloud-images"
```

Comment:
I used to have these inside the script. I moved them outside to keep the script kinda generic but since `inlineScript` runs inside a container, I'm not sure now if they are also set inside the container. Need to trigger the pipeline again to check. Worst case, we move them back in the script.

Reply:
It works. All good.